pyoverkiz/models.py

```diff
@@ -271,6 +271,30 @@ def __init__(
         self.value = value
         self.type = DataType(type)
 
+    @property
+    def value_as_int(self) -> int:
```

Comment:
Does this take into account None values? Or are they not possible?
```suggestion
    def value_as_int(self) -> int | None:
```

Reply:
I don’t remember a case where the API returned a state without any value. @iMicknl ? There was not also a DataType.NONE?

Reply:
I saw a few None checks inside the HA code, but I didn't look too deep.
I thought maybe something like this was safer
```python
        if self.type == DataType.NONE:
            return None
        if self.type == DataType.INTEGER:
            return cast(int, self.value)
        raise TypeError(f"{self.name} is not an integer")
```

Reply:
self.value can be None, this is the case when `DataType.None`
https://github.com/iMicknl/python-overkiz-api/blob/4d9745f9c272f96d05a60a9e49e577894206648f/pyoverkiz/models.py#L267

We also have an example of that in our test fixtures.

Reply:
@iMicknl `DataType.None` does not exist. I double check within TahomaLink website source, and they still haven’t:

```
        switch (t) {
          case DataType.INTEGER:
            return Number.parseInt(e);
          case DataType.FLOAT:
            return Number.parseFloat(e);
          case DataType.BOOLEAN:
            return r(e);
          case DataType.STRING:
          case DataType.BLOB:
            return e;
          case DataType.DATE:
          case DataType.JSON_OBJECT:
          case DataType.JSON_ARRAY:
            return "string" == typeof e ? JSON.parse(e) : e;
        }
```

Reply:
@tetienne see https://github.com/iMicknl/python-overkiz-api/blob/24b9bd708f6d0dd7b2f1846dec15a45e61e9c293/tests/events.json#L30-L34 for an example of DataType 0. (value is missing).

Reply:
Thx I missed this one:
https://github.com/iMicknl/python-overkiz-api/blob/24b9bd708f6d0dd7b2f1846dec15a45e61e9c293/pyoverkiz/enums/general.py#L21

So we are fine, Overkize indicate if there is a value or not. Meaning if we DataType.INTEGER (or whatever), we are sure to get a value.

Reply:
My only concern is that it seems that this can change. `core:ConsumptionTariff4State` is an int normally, however for this devcice the state is empty. So shouldn't we now always need to check if the state is not empty?

Reply:
I see what you mean. So you want I returned None in such case as suggested by @epenet?

Reply:
I think we just need to see how this current change works with Overkiz in core (just in a dev branch) and than make changes accordingly.

Reply:
OK I’ve implemented @epenet suggestion. So it will force the client to check the returned value. 

About the method names, it’s ok? I hesitated between `int_value` and `value_as_int`.